cmd/etcd.go

```diff
@@ -0,0 +1,81 @@
+and starts an embedded etcd.
+
+Flags:
+	--backup-restore-tls-enabled
+		Enables TLS for communicating with backup-restore if its value is true. It is disabled by default.
+	--backup-restore-host-port
+		Host address and port of the backup restore with which this container will interact during initialization. Should be of the format <host>:<port> and must not include the protocol.
+	--backup-restore-ca-cert-bundle-path
+		Path of CA cert bundle (This will be used when TLS is enabled via backup-restore-tls-enabled flag.
+    --etcd-client-cert-path
+		Path of TLS certificate of the etcd client (This will be used if client-transport-security is set in the etcd configuration).
+	--etcd-client-key-path
+		Path of TLS key of the etcd client (This will be used if client-transport-security is set in the etcd configuration).
+	--etcd-server-name
+		Name of the server (host) which will be used to configure TLS config to connect to the etcd server process.
+	--etcd-ready-timeout
+		time duration the application will wait for etcd to get ready, by default it waits forever.`,
+		AddFlags: AddEtcdFlags,
+		Run:      InitAndStartEtcd,
+	}
+	config = types.Config{}
+	// etcdReadyTimeout is the timeout for an embedded etcd server to be ready.
+	etcdReadyTimeout time.Duration
+)
+
+// AddEtcdFlags adds flags from the parsed FlagSet into application structs
+func AddEtcdFlags(fs *flag.FlagSet) {
+	fs.BoolVar(&config.BackupRestore.TLSEnabled, "backup-restore-tls-enabled", types.DefaultBackupRestoreTLSEnabled, "Enables TLS for communicating with backup-restore container")
+	fs.StringVar(&config.BackupRestore.HostPort, "backup-restore-host-port", types.DefaultBackupRestoreHostPort, "Host and Port to be used to connect to the backup-restore container")
+	fs.StringVar(&config.BackupRestore.CaCertBundlePath, "backup-restore-ca-cert-bundle-path", "", "File path of CA cert bundle to help establish TLS communication with backup-restore container")
+	fs.StringVar(&config.EtcdClientTLS.ServerName, "etcd-server-name", "", "Name of the server (host) which will be used to configure TLS config to connect to the etcd server process")
+	fs.StringVar(&config.EtcdClientTLS.CertPath, "etcd-client-cert-path", "", "File path of ETCD client certificate to help establish TLS communication of the client to ETCD")
+	fs.StringVar(&config.EtcdClientTLS.KeyPath, "etcd-client-key-path", "", "File path of ETCD client key to help establish TLS communication of the client to ETCD")
+	fs.DurationVar(&etcdReadyTimeout, "etcd-ready-timeout", 0, "Time duration to wait for etcd to be ready")
+}
+
+// InitAndStartEtcd sets up and starts an embedded etcd
+func InitAndStartEtcd(ctx context.Context, cancelFn context.CancelFunc, logger *zap.Logger) error {
+	etcdApp, err := app.NewApplication(ctx, cancelFn, config, etcdReadyTimeout, logger)
+	if err != nil {
+		return err
+	}
+	if err := etcdApp.Setup(); err != nil {
+		return err
+	}
+	return etcdApp.Start()
```

Comment:
No, 2379 for client requests and 2380 for peer communication will be used by default. I think even today it is not possible to change these ports via any exposed configuration. Do you have a specific use case because of which you wish to have these configurable?
IANA has officially allocated these ports for etcd (http://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=etcd). So i do not find any reason to make this configurable at this point in time.